core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -220,7 +219,7 @@ public boolean hasDerivedDependencyErrors() {
      * The core can depend on a plugin if it is bundled. Sometimes it's the only thing that
      * depends on the plugin e.g. UI support library bundle plugin.
      */
-    private static Set<String> CORE_ONLY_DEPENDANT = ImmutableSet.copyOf(Collections.singletonList("jenkins-core"));
+    private static Set<String> CORE_ONLY_DEPENDANT = Collections.unmodifiableSet(new HashSet<>( Arrays.asList("jenkins-core")));
```

Comment:
Simpler:

```suggestion
    private static Set<String> CORE_ONLY_DEPENDANT = Collections.singleton("jenkins-core");
```

[Collections#singleton](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singleton-T-) returns an immutable set.